src/components/Cities.js

```diff
@@ -1,20 +1,32 @@
 import React, { useEffect, useState } from 'react';
 import {
-  Loader, Text, Image,
+  Loader, Text, Image, Input,
 } from '@mantine/core';
 import { useSelector } from 'react-redux';
+import { FaSearch } from 'react-icons/fa';
 import CityItem from './CityItem';
 import './cities.css';
 
 const Cities = () => {
+  const { cities } = useSelector((state) => state.cities);
   const [loading, setLoading] = useState(true);
+  const [filteredData, setFilteredData] = useState([]);
+
   useEffect(() => {
-    setTimeout(() => {
+    setLoading(true);
+    // setTimeout(() => {
+    if (cities.length > 0) {
       setLoading(false);
-    }, 500);
-  }, []);
+    }
+    setFilteredData(cities);
+    // }, 1000);
+  }, [cities]);
 
-  const { cities } = useSelector((state) => state.cities);
+  const handleChange = (text) => {
+    const content = text.target.value;
+    console.log(content);
+    setFilteredData(cities.filter((e) => e.city.toLowerCase().includes(content.toLowerCase())));
+  };
 
```

Comment:
- [ ] Dear @chiztechnology, Good job making the requested changes. But while doing this, I notice you have commented on old codes. And it's among the primary requirement to keep your codebase clean. Please consider removing commented codes as it will make your codebase messy, Remember if you ever like to see the codes you have written once you can use your commit history. Check this list of [JavaScript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md) for more on this.
- [Optional] Also, I notice your app works fine locally but fails on the live server. The last reviewer mention not having a place to search for items that could be filtered. Now you have addressed the search box ✅. Please notice your detailed page is currently not loading any data for a specific area. This is among the primary requirements and I strongly suggest you address it. 
> As I see from the error in the console this is caused by using an `HTTP` instead of `HTTPS`, Please consider using `https://api.weatherapi.com/v1/current.json` instead of `http://api.weatherapi.com/v1/current.json`

https://user-images.githubusercontent.com/69077061/221374671-bd3e3017-4dac-4989-a99f-d53289841480.mp4

Reply:
hello, dear thanks a lot for your great review and your amazing video.
I really appreciate the fact you took the time to record this video explaining the behavior of the website in the browser.

The app works fine locally, but the hosted version is failing to fetch each weather detail due to some CORS Policy.
kindly have a look at the logs below. 👇 
<img width="1440" alt="Screenshot 2023-02-25 at 22 12 03" src="https://user-images.githubusercontent.com/68948316/221377728-bcebddef-0a4d-445c-8d90-1463e3d7a556.png">

I cleared the codebase as requested by removing all unnecessary commented code.